src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import {
+  View,
+  ScrollView,
+  Image,
+  ViewProps,
+} from 'react-native';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import {
+  RadioGroup,
+  Radio,
+  RadioProps,
+  Button,
+  Input,
+} from '@kitten/ui';
+import { Text } from '@src/components/common';
+import {
+  Product as ProductModel,
+  Comment as CommentModel,
+} from '@src/core/model';
+import { CommentsList1 } from '@src/containers/layouts/messaging/commentsList1/commentsList1.component';
+
+interface ComponentProps {
+  product: ProductModel;
+  comments: CommentModel[];
+  currentCommentText: string;
+  onCommentSubmit: () => void;
+  onCommentTextChange: (text: string) => void;
+  onCommentItemPress: (index: number) => void;
+  onCommentPress: (index: number) => void;
+  onLikePress: (index: number) => void;
+  onBuyPress: () => void;
+}
+
+interface State {
+  selectedColorIndex: number;
+}
+
+export type ProductDetailsProps = ThemedComponentProps & ComponentProps;
+
+class ProductDetailsComponent extends React.Component<ProductDetailsProps, State> {
+
+  public state: State = {
+    selectedColorIndex: 0,
+  };
+
+  private onBuyPress = (): void => {
+    this.props.onBuyPress();
+  };
+
+  private onSelectColorIndex = (selectedColorIndex: number): void => {
+    this.setState({ selectedColorIndex });
+  };
+
+  private onCommentButtonPress = (index: number) => {
+    this.props.onCommentPress(index);
+  };
+
+  private onLikeButtonPress = (index: number) => {
+    this.props.onLikePress(index);
+  };
+
+  private onMoreButtonPress = (index: number) => {
+    this.props.onCommentItemPress(index);
+  };
+
+  private onCommentTextChange = (text: string): void => {
+    this.props.onCommentTextChange(text);
+  };
+
+  private handleTextSubmit = () => {
+    this.props.onCommentSubmit();
+  };
+
+  private renderColorRadio = (color: string, index: number): React.ReactElement<RadioProps> => {
+    const { themedStyle } = this.props;
+
+    return (
+      <Radio
+        style={themedStyle.radioItem}
+        key={index}
+        text={color}
+      />
+    );
+  };
+
+  private renderInfoContainer = (): React.ReactElement<ViewProps> => {
```

Comment:
prefer refactoring `render` methods to components even if it's used once

Reply:
I don't see any sense in that. The reason of creating the separate component - only when it'll be reused multiple times.